src/core/VectorGraphModel.cpp

```diff
@@ -0,0 +1,2411 @@
+			}
+
+			diff = diff * diff * 10.0f * direction;
+			diff = std::clamp(diff, -0.7f, 0.7f);
+			m_dataArray[i - 1].m_c = diff;
+
+			noneBefore = diff < 0.1f && diff > -0.1f;
+			isNegativeBefore = isNegative;
+		}
+	}
+	// the whole m_dataArray needs to be updated
+	getUpdatingFromPoint(-1);
+	if (callDataChanged == true)
+	{
+		dataChanged();
+	}
+}
+void VectorGraphDataArray::setDataArray(std::vector<float>* inputDataArray,
+	bool shouldCurve, bool shouldClear, bool shouldClamp, bool shouldRescale, bool callDataChanged)
+{
+	std::vector<PointF> convertedDataArray(inputDataArray->size());
+	float stepSize = 1.0f / static_cast<float>(convertedDataArray.size());
+	for (unsigned int i = 0; i < inputDataArray->size(); i++)
+	{
+		convertedDataArray[i].first = i * stepSize;
+		convertedDataArray[i].second = (*inputDataArray)[i];
+	}
+	setDataArray(&convertedDataArray, shouldCurve, shouldClear, shouldClamp, shouldRescale, false, callDataChanged);
+}
+void VectorGraphDataArray::setDataArray(float* inputDataArray, unsigned int size,
+	bool shouldCurve, bool shouldClear, bool shouldClamp, bool shouldRescale, bool callDataChanged)
+{
+	std::vector<PointF> convertedDataArray(size);
+	float stepSize = 1.0f / static_cast<float>(size);
+	for (unsigned int i = 0; i < size; i++)
+	{
+		convertedDataArray[i].first = i * stepSize;
+		convertedDataArray[i].second = inputDataArray[i];
+	}
+	setDataArray(&convertedDataArray, shouldCurve, shouldClear, shouldClamp, shouldRescale, false, callDataChanged);
+}
+
+unsigned int VectorGraphDataArray::setX(unsigned int pointLocation, float newX)
+{
+	int location = pointLocation;
+	if (m_isFixedX == false && newX <= 1.0f)
+	{
+		bool found = false;
+		bool isBefore = false;
+		location = getNearestLocation(newX, &found, &isBefore);
+		// if an other point was not found exactly at newX
+		// and if dataArray end points are changeable
+		if (found == false && ((m_isFixedEndPoints == true &&
+			pointLocation < m_dataArray.size() - 1 && pointLocation > 0) ||
+			m_isFixedEndPoints == false))
+		{
+			int targetLocation = location;
+			// bool dataChangedVal = false;
+			// if getNearestLocation returned a value
+			if (location >= 0)
+			{
+				if (location < static_cast<int>(pointLocation) && isBefore == true)
+				{
+					if (targetLocation + 1 < static_cast<int>(m_dataArray.size()))
+					{
+						targetLocation++;
+					}
+				}
+				else if (location > static_cast<int>(pointLocation) && isBefore == false)
+				{
+					if (targetLocation > 0)
+					{
+						targetLocation--;
+					}
+				}
+				m_dataArray[pointLocation].m_x = newX;
+				swap(pointLocation, targetLocation, true);
+				location = targetLocation;
+
+				getUpdatingFromPoint(-1);
+				// changes in the position can change lines before
+				// so the point before this is updated
+				if (location > 0)
+				{
+					getUpdatingFromPoint(location - 1);
+				}
+				dataChanged();
+			}
+			else
+			{
+				location = pointLocation;
```

Comment:
if you decide on inverting the conditions, you can return pointLocation directly in these 2 areas.